src/box/lua/console.c

```diff
@@ -760,22 +760,27 @@ lua_rl_dmadd(dmlist *ml, const char *p, size_t pn, const char *s, int suf)
 	return 0;
 }
 
-/* Get __index field of metatable of object on top of stack. */
+/* Get a table from __autocomplete method of metatable if present. */
 static int
-lua_rl_getmetaindex(lua_State *L)
+lua_rl_getcompletetab(lua_State *L)
 {
 	if (!lua_getmetatable(L, -1)) {
 		lua_pop(L, 1);
 		return 0;
 	}
-	lua_pushstring(L, "__index");
+	lua_pushstring(L, "__autocomplete");
```

Comment:
We should fallback on `__index` if `__autocomplete` isn't defined. If we don't, autocomplete stops working in some cases, e.g. for `box.space.test:replace`. Please fix and remove useless `__autocomplete` methods added by this patch (the ones that return `__index`).